[george-yeo] iP

george-yeo opened the pull request:
# Oyster
> The world is your _oyster_

Oyster Chatbot
- User-experience first
- Productivity
- Easy to pick up 😉

## Features
1. Task Management
2. Deadline Management [Coming soon]
3. Reminders [Coming soon]

## Download
- [x] Download [here](https://github.com/george-yeo/ip)!

Use these methods to quickly load Oyster anywhere!
```
oyster.begin()
oyster.close()
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

george-yeo committed:
done up to level 2

george-yeo committed:
done up to level 3

george-yeo committed:
up to level 6

george-yeo committed:
updated test cases

george-yeo committed:
Refactor TodoTask class

george-yeo committed:
Refactor DisplayHandler class

george-yeo committed:
Add Level 7

george-yeo committed:
Add Level 8

george-yeo committed:
Rename Display and FileHandler

george-yeo committed:
Edit DateFormatException

george-yeo committed:
Merge remote-tracking branch 'origin/add-gradle-support'

george-yeo committed:
Add Gradle

george-yeo committed:
Add JUnit

george-yeo committed:
Add JUnit

george-yeo committed:
Update build.gradle

george-yeo committed:
Add JavaDoc

george-yeo committed:
Add FindCommand

george-yeo committed:
Merge branch 'branch-A-JavaDoc'

george-yeo committed:
Edit JavaDocs

george-yeo committed:
Merge branch 'branch-A-JavaDoc'

george-yeo committed:
Add checkstyle

george-yeo committed:
Edit style

george-yeo committed:
Add assertions

To prevent programmer code errors in the flow of program,
assertions are added to check for null inputs, among other checks.

george-yeo committed:
Add GUI

Johnwz123 commented on src/main/java/oyster/commands/Command.java:
```diff
@@ -0,0 +1,50 @@
+ * Command Class that holds an executable callback.
+ */
+public abstract class Command {
+    private String[] message;
```
Plural form should be used on names representing a collection of objects.
```suggestion
    private String[] messages;
```

Johnwz123 commented on src/main/java/oyster/tasks/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+    }
+
+    /**
+     * Insert a Task into the list.
```
```suggestion
     * Inserts a Task into the list.
```

Johnwz123 commented on src/main/java/oyster/tasks/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+    }
+
+    /**
+     * Delete a Task in the list.
```
```suggestion
     * Deletes a Task in the list.
```
Do make the same change for the subsequent comments in this file.

Johnwz123 commented on src/main/java/oyster/utils/Parser.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    /**
+     * Convert TaskList into a String.
```
```suggestion
     * Converts TaskList into a String.
```

Johnwz123 reviewed (commented):
A few nits here and there

elliot-tang commented on src/main/java/oyster/tasks/DeadlineTask.java:
```diff
@@ -0,0 +1,85 @@
+     * @param input The string to parse into a DeadlineTask.
+     * @return DeadlineTask object.
+     */
+    public static DeadlineTask fromInput(String input) {
+        String name = "";
+        String deadline = "";
+        boolean isDeadlineNext = false;
```
perhaps you could consider renaming this method to createFromInput, to signpost that this method creates a task. similarly for the other task types.

elliot-tang commented on src/main/java/oyster/tasks/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+     *
+     * @return A copy of the ArrayList used to hold all the Tasks.
+     */
+    public ArrayList<Task> getItems() {
+        @SuppressWarnings("unchecked")
+        ArrayList<Task> copy = (ArrayList<Task>) tasks.clone();
+        return copy;
+    }
```
perhaps this method could be renamed to getTasks or getCopyOfTasks, which would be more specific than getItems

elliot-tang commented on src/main/java/oyster/tasks/EventTask.java:
```diff
@@ -0,0 +1,108 @@
+     * @param input The string to parse into a EventTask.
+     * @return EventTask object.
+     */
+    public static EventTask fromInput(String input) {
+        String name = "";
+        String from = "";
+        String to = "";
+        int isDateCounter = 0;
```
perhaps this can be changed to createFromInput, as previously mentioned

elliot-tang commented on src/main/java/oyster/tasks/ToDoTask.java:
```diff
@@ -0,0 +1,49 @@
+     * @param input The string to parse into a ToDoTask.
+     * @return ToDoTask object.
+     */
+    public static ToDoTask fromInput(String input) {
+        input = input.trim();
+
+        if (input.isEmpty()) {
+            throw new TaskFieldException("Description");
+        }
+
+        return new ToDoTask(input);
+    }
```
perhaps this can be changed to createFromInput, as previously mentioned

elliot-tang reviewed (commented):
LGTM! great job with the namings overall, code and names are easy to understand, just some nits to consider.

george-yeo committed:
Refactor code

george-yeo committed:
Add streams

george-yeo committed:
Clean code

george-yeo committed:
Clean code

george-yeo committed:
Merge pull request #3 from george-yeo/branch-A-CodeQuality

Refactor code

george-yeo committed:
Merge branch 'branch-A-Assertions'

# Conflicts:
#	src/main/java/oyster/tasks/DeadlineTask.java
#	src/main/java/oyster/tasks/EventTask.java
#	src/main/java/oyster/tasks/ToDoTask.java
#	src/main/java/oyster/ui/MainWindow.java
#	src/main/resources/images/DaDuke.png
#	src/main/resources/images/DaUser.png
#	src/main/resources/view/DialogBox.fxml
#	src/main/resources/view/MainWindow.fxml

george-yeo committed:
Merge branch 'branch-A-Streams'

george-yeo committed:
Fix Ui

george-yeo committed:
Fix style

george-yeo committed:
Add reminders

george-yeo committed:
Add UserGuide

Added a README detailing how to use each command in the chatbot, and
their respective expected outputs.

george-yeo committed:
Improve code quality
